internal/storage/v1/elasticsearch/spanstore/readerv1.go

```diff
@@ -44,10 +44,10 @@ func (s *SpanReaderV1) GetTrace(ctx context.Context, query spanstore.GetTracePar
 	return &model.Trace{Spans: spans}, nil
 }
 
-func (s *SpanReaderV1) collectSpans(dbSpans []*dbmodel.Span) ([]*model.Span, error) {
+func (s *SpanReaderV1) collectSpans(dbSpans []dbmodel.Span) ([]*model.Span, error) {
 	spans := make([]*model.Span, len(dbSpans))
 	for i, jsonSpan := range dbSpans {
-		span, err := s.spanConverter.SpanToDomain(jsonSpan)
+		span, err := s.spanConverter.SpanToDomain(&jsonSpan)
```

Comment:
Defeats the purpose of avoiding pointers - value escapes to heap here.

`jsonSpan` is forced to be a copy. Instead do a loop with index only, and use `&dbSpans[i]` to avoid allocating.